### DISCUSSION TITLE
Can't build simple HarvestProject example

### DISCUSSION THREAD

**Discussion by amtopel** [wixtoolset/issues#8757] [Questions]
Hi, I'm unable to build the simple `HarvestProject` example available [here](https://wixtoolset.org/docs/tools/heat/#using-harvestproject-to-harvest-output-from-a-project).  I've set up a bare-bones Wix installer project using the Visual Studio Heatwave extension.  And I'm trying to harvest the output from a neighboring C++ project.  I encounter two alternative problems: either I get a `WIX0094` error (identifier could not be found), or I get a `WIX0091` error (Duplicate WixComponentGroup).

My first attempt is the linked example, essentially verbatim.  This my project file:

```
<Project Sdk="WixToolset.Sdk/5.0.1">
	
	<PropertyGroup>
		<EnableProjectHarvesting>true</EnableProjectHarvesting>
		<HarvestProjectsSuppressUniqueIds>true</HarvestProjectsSuppressUniqueIds>
	</PropertyGroup>

	<ItemGroup>
		<HarvestProject Include="..\TestProject\TestProject.vcxproj" ProjectOutputGroups="Binaries" />
	</ItemGroup>

	<ItemGroup>
		<ProjectReference Include="..\TestProject\TestProject.vcxproj" DoNotHarvest="true"/>
	</ItemGroup>

	<ItemGroup>
		<PackageReference Include="WixToolset.Heat" />
	</ItemGroup>

</Project>
```

And this is my Package.wxs file:

```
<Wix xmlns="http://wixtoolset.org/schemas/v4/wxs">
  <Package Name="Installer" Manufacturer="TODO Manufacturer" Version="1.0.0.0" UpgradeCode="33810ee6-5c7e-492a-8184-239143dbc965">
    <MajorUpgrade DowngradeErrorMessage="!(loc.DowngradeError)" />

    <Feature Id="Main">
	<ComponentGroupRef Id="TestProject.Binaries" />
    </Feature>
	  
  </Package>
</Wix>
```

This gives the `WIX0094` error: "The identifier 'Directory:TestProject.Binaries' could not be found. Ensure you have typed the reference correctly and that all the necessary inputs are provided to the linker."

I can fix this error by removing the `DoNotHarvest="true"` in the project file.  But then I get the `WIX0091` error: "Duplicate WixComponentGroup with identifier 'TestProject.Binaries' . . ."	

Can anyone explain what I'm doing wrong?  Thanks a lot.

**Comment by jb455** [comment]
Not sure if this fully answers your question but I was stuck on harvesting for a while too and eventually found that there's a new way to do it in V5, documented here: https://wixtoolset.org/docs/fivefour/#file-harvesting. The way I have it working in my project is like this (inside the `Package` tags):
```
<?define MainOutputDir = $(var.SolutionDir)/[App]/bin/$(var.Configuration)/net8.0-windows ?>
<Feature Id="App">
	<!-- Include all files from output folder -->
	<Files Include="$(var.MainOutputDir)/**" Directory="INSTALLFOLDER" />
</Feature>
```
